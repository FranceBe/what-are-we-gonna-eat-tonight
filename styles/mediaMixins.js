/* eslint-disable no-param-reassign */

import { css } from 'styled-components'

export const minSizes = {
  sm: 576,
  md: 768,
  lg: 992,
  xl: 1200,
  xxl: 1600,
  xxxl: 2100,
}
export const mediaMin = Object.keys(minSizes).reduce((acc, label) => {
  acc[label] = arg => css`
    @media (min-width: ${minSizes[label] / 16}em) {
      ${arg}
    }
  `
  return acc
}, {})

const maxSizes = {
  xs: 575,
  sm: 767,
  md: 991,
  lg: 1199,
  xl: 1599,
  xxl: 2199,
}
export const mediaMax = Object.keys(maxSizes).reduce((acc, label) => {
  acc[label] = arg => css`
    @media (max-width: ${maxSizes[label] / 16}em) {
      ${arg}
    }
  `
  return acc
}, {})

const onlySizes = {
  xs: {
    max: maxSizes.sm,
  },
  sm: {
    min: minSizes.sm,
    max: maxSizes.md,
  },
  md: {
    min: minSizes.md,
    max: maxSizes.lg,
  },
  lg: {
    min: minSizes.lg,
    max: minSizes.xl,
  },
  xl: {
    min: minSizes.xl,
  },
}

function returnMinMedia(size) {
  return size ? `(min-width: ${size / 16}em)` : ''
}

function returnMaxMedia(size) {
  return size ? `(max-width: ${size / 16}em)` : ''
}

function returnAnd(minSize, maxSize) {
  return minSize && maxSize ? ' and ' : ''
}

export const mediaOnly = Object.keys(onlySizes).reduce((acc, label) => {
  acc[label] = arg => css`
    @media ${returnMinMedia(
    onlySizes[label].min
  )}${returnAnd(onlySizes[label].min, onlySizes[label].max)}${returnMaxMedia(
  onlySizes[label].max
)} {
      ${arg}
    }
  `
  return acc
}, {})
