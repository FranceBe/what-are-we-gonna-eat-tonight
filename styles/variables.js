export const palette = {
  white: '#FFF',
  orange_white: '#FEFEF1',
  tacao: '#EBA789',
  kabul: '#5F4C44',
  apple: '#50C92C',
  lemon: '#F6FA19',
  burnt_sienna: '#EE6663',
}

export const colors = {
  primary_light: palette.orange_white,
  primary_default: palette.tacao,
  primary_dark: palette.kabul,
  neutral_lightest: palette.white,
  secondary_default: palette.apple,
  secondary_light: palette.lemon,
  tertiary_default: palette.burnt_sienna,
}

export const fontSizes = {
  px_14: '0.875rem',
  px_20: '1.25rem',
  px_30: '1.875rem',
}

export const spaces = {
  xtraSmall: '1px',
  small: '5px',
  medium: '10px',
  regular: '20px',
  regular_large: '30px',
  large: '50px',
  xtraLarge: '200px',
  xxtraLarge: '300px',
}
