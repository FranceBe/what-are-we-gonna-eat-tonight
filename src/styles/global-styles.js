import { createGlobalStyle } from 'styled-components'
import { colors, spaces } from 'styles/variables'

// PlayfairDisplay
import PlayfairDisplaytRegular from 'assets/fonts/PlayfairDisplay-Regular.ttf'
import PlayfairDisplayBold from 'assets/fonts/PlayfairDisplay-Bold.ttf'

export const GlobalStyle = createGlobalStyle`

  @font-face {
    font-family: 'PlayfairDisplay';
    src: url(${PlayfairDisplaytRegular}) format('ttf');
    font-style: normal;
    font-weight: 400;
    font-display: swap;
  }

  @font-face {
    font-family: 'PlayfairDisplay';
    src: url(${PlayfairDisplayBold}) format('ttf');
    font-style: normal;
    font-weight: 700;
    font-display: swap;
  }

  html, body {
    -webkit-font-smoothing: antialiased;
    -moz-font-smoothing: antialiased;
    -o-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    height: 100%;
  }
  body {
    background-color: ${colors.primary_light};
    margin: ${spaces.small};
    color: ${colors.primary_dark};
    font-family: 'PlayfairDisplay', sans-serif;
  }
  `
