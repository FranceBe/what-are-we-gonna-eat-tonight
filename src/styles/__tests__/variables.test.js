import { spaces, colors, fontSizes, palette } from 'styles/variables'

describe('variables', () => {
  describe('spaces', () => {
    it('should match snapshot', () => {
      expect(spaces).toMatchSnapshot()
    })
  })
  describe('colors', () => {
    it('should match snapshot', () => {
      expect(colors).toMatchSnapshot()
    })
  })
  describe('fontSizes', () => {
    it('should match snapshot', () => {
      expect(fontSizes).toMatchSnapshot()
    })
  })
  describe('palette', () => {
    it('should match snapshot', () => {
      expect(palette).toMatchSnapshot()
    })
  })
})
