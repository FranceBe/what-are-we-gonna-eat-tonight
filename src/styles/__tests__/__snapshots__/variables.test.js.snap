// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`variables colors should match snapshot 1`] = `
Object {
  "neutral_lightest": "#FFF",
  "primary_dark": "#5F4C44",
  "primary_default": "#EBA789",
  "primary_light": "#FEFEF1",
  "secondary_default": "#50C92C",
  "secondary_light": "#F6FA19",
  "tertiary_default": "#EE6663",
}
`;

exports[`variables fontSizes should match snapshot 1`] = `
Object {
  "px_14": "0.875rem",
  "px_20": "1.25rem",
  "px_30": "1.875rem",
}
`;

exports[`variables palette should match snapshot 1`] = `
Object {
  "apple": "#50C92C",
  "burnt_sienna": "#EE6663",
  "kabul": "#5F4C44",
  "lemon": "#F6FA19",
  "orange_white": "#FEFEF1",
  "tacao": "#EBA789",
  "white": "#FFF",
}
`;

exports[`variables spaces should match snapshot 1`] = `
Object {
  "large": "50px",
  "medium": "10px",
  "regular": "20px",
  "regular_large": "30px",
  "small": "5px",
  "xtraLarge": "200px",
  "xtraSmall": "1px",
  "xxtraLarge": "300px",
}
`;
